### DISCUSSION TITLE
In between sea-query and sea-orm

### DISCUSSION THREAD

**Discussion by Sytten** [SeaQL/sea-query#245] [Ideas]
I was looking back at a startup I worked on for a while that was coded in golang.
I used this very good query builder called go-jet (https://github.com/go-jet/jet).

What I liked about it now is that is kinda fits in between where sea-query is at right now and where sea-orm is.
A typical select query would look like:
```golang
SELECT(
    Payment.AllColumns,
    Customer.AllColumns,
).
FROM(Payment.
    INNER_JOIN(Customer, Payment.CustomerID.EQ(Customer.CustomerID))).
ORDER_BY(Payment.PaymentID.ASC()).
LIMIT(30)
```
As you can see it is not far from what sea-query offers but has a few benefits like not having to specify the table name and `AllColumns`

It also has the benefit of doing result to model mapping automatically and generated the models based on database schema (similar to sea-orm).

I feel this could be a third way for people that might never want the full ORM but want to have a better DX with the query builder.

**Comment by billy1624** [comment]
Hey @Sytten, this is interesting. Actually, @tqwewe got a working PR SeaQL/sea-orm#241 that could generate entity files at compile-time

**Reply by Sytten** [reply]
This different I would say. What I am proposing would be kind of a type aware query builder.
It is a query builder in the sense that you have a to write SQL but it still uses models that are typed. You get a runtime error if the select doesn't match the data type you try to use.
